src/pages/List.js

```diff
@@ -77,27 +136,125 @@ export default function List({ token }) {
+      (doc) =>
+        doc
+          .data()
+          .item_name.toLowerCase()
+          .includes(query.toLowerCase().trim()) || query === '',
+    );
+  };
+
+  const filterByLessThanSevenDays = (listItems) => {
+    // filter the items by user input or render all items if no input
+    const alphabetizedUserInputOrAllItems = filterByUserInput(listItems);
+
+    // filter into the green category of less than 7 days
+    return alphabetizedUserInputOrAllItems.filter((item) => {
+      if (item.data().times_purchased === 0) {
+        return item.data().purchase_frequency === 7;
+      } else {
+        return item.data().last_estimate < 7 && !checkForInactiveItem(item);
+      }
+    });
+  };
+
+  const filterByMoreThanSevenDaysAndLessThanThirtyDays = (listItems) => {
+    // filter the items by user input or render all items if no input
+    const alphabetizedUserInputOrAllItems = filterByUserInput(listItems);
+
+    // filter items into the purple category of more than 7 days and less than 30 days
+    return alphabetizedUserInputOrAllItems.filter((item) => {
+      if (item.data().times_purchased === 0) {
+        return item.data().purchase_frequency === 14;
+      } else {
+        return (
+          item.data().last_estimate >= 7 &&
+          item.data().last_estimate <= 30 &&
+          !checkForInactiveItem(item)
+        );
+      }
+    });
+  };
+
+  const filterByMoreThanThirtyDays = (listItems) => {
+    // filter the items by user input or render all items if no input
+    const alphabetizedUserInputOrAllItems = filterByUserInput(listItems);
+
+    // filter items into the red category of more than 30 days
+    return alphabetizedUserInputOrAllItems.filter((item) => {
+      if (item.data().times_purchased === 0) {
+        return item.data().purchase_frequency === 30;
+      } else {
+        return item.data().last_estimate > 30 && !checkForInactiveItem(item);
+      }
+    });
+  };
+
+  const filterByInactiveItems = (listItems) => {
+    // filter the items by user input or render all items if no input
+    const alphabetizedUserInputOrAllItems = filterByUserInput(listItems);
+
+    // filter items into the gray category of more than double last_estimate
+    return alphabetizedUserInputOrAllItems.filter((item) =>
+      checkForInactiveItem(item),
+    );
+  };
+
+  const renderUnorderedList = (doc, color) => {
+    return (
+      <li key={doc.id} className="checkbox-wrapper" style={{ color: color }}>
+        <input
+          type="checkbox"
+          id={doc.id}
+          defaultChecked={compareTimeStamps(doc.data().last_purchased)}
+          disabled={compareTimeStamps(doc.data().last_purchased)}
+          onClick={(e) => markItemPurchased(e, doc.id, doc.data())}
+        />
+        <label htmlFor={doc.id}>{doc.data().item_name}</label>
+        <button key={doc.id} onClick={() => deleteItem(doc.id)}>
+          Delete
+        </button>
+      </li>
+    );
+  };
```

Comment:
This kind of extraction to reduce duplication is great! It's clear reading this what its operands are and what will result! 👏